tests/test_chiral_restraints.py

```diff
@@ -687,92 +687,62 @@ def test_chiral_inversion_in_single_topology_runs(well_aligned):
     _ = run_vacuum(atom_map.mol_a, atom_map.mol_b, atom_map.core, ff, None, very_short_hrex_params, n_windows=3)
 
 
-# some util fxns: TODO: move these where appropriate
-def make_chiral_restr_fxns(mol_a, mol_b, chiral_k=1000.0):
-    from timemachine.fe.chiral_utils import setup_all_chiral_atom_restr_idxs
-
-    restr_idxs_a = np.array(setup_all_chiral_atom_restr_idxs(mol_a, utils.get_romol_conf(mol_a)))
-    restr_idxs_b = np.array(setup_all_chiral_atom_restr_idxs(mol_b, utils.get_romol_conf(mol_b)))
-
-    @jit
-    def U_a(x_a):
-        return U_chiral_atom_batch(x_a, restr_idxs_a, chiral_k).sum()
-
-    @jit
-    def U_b(x_b):
-        return U_chiral_atom_batch(x_b, restr_idxs_b, chiral_k).sum()
-
-    return U_a, U_b
-
-
-# TODO: should def x_ab_from_x, def xs_ab_from_xs move to a utils.py?
-def x_ab_from_x(x, atom_map):
-    """extract confs of mol_a, mol_b from combined conf
-
-    TODO: jaxify this...
-    """
-    # duplicating from https://github.com/proteneer/timemachine/blob/85c15ac58315e7304233ac847fe2d24db8c1773f/timemachine/fe/cif_writer.py#L9
-    n_a = atom_map.mol_a.GetNumAtoms()
-    n_b = atom_map.mol_b.GetNumAtoms()
-
-    x_a = np.zeros((n_a, 3))
-    x_b = np.zeros((n_b, 3))
-
-    for a_idx, c_idx in enumerate(atom_map.a_to_c):
-        x_a[a_idx] = x[c_idx]
-    for b_idx, c_idx in enumerate(atom_map.b_to_c):
-        x_b[b_idx] = x[c_idx]
-    #
-
-    return x_a, x_b
-
-
-def xs_ab_from_xs(xs, atom_map):
-    """map x_ab_from_x over xs
-
-    TODO: remove when x_ab_from_x is jaxified
-    """
-    xs_a, xs_b = [], []
-    for x in xs:
-        x_a, x_b = x_ab_from_x(x, atom_map)
-        xs_a.append(x_a)
-        xs_b.append(x_b)
-    xs_a = np.array(xs_a)
-    xs_b = np.array(xs_b)
-    return xs_a, xs_b
+@pytest.mark.nightly(reason="slow")
+@pytest.mark.parametrize("well_aligned", [True, False])
+def test_chiral_inversion_in_single_topology(well_aligned):
+    """assert chiral consistency preserved through vacuum HREX"""
 
+    ff = Forcefield.load_default()
 
-def make_chiral_flip_heatmaps(vacuum_results, atom_map):
-    """evaluate mol_a and mol_b chiral restraint energy in each frame of every trajectory"""
-    mol_a_chiral_conflicts = []
-    mol_b_chiral_conflicts = []
-    U_a, U_b = make_chiral_restr_fxns(atom_map.mol_a, atom_map.mol_b)
+    atom_map = make_chiral_flip_pair(well_aligned)
 
-    for traj in vacuum_results.frames:
-        xs = np.array(traj)
-        xs_a, xs_b = xs_ab_from_xs(np.array(xs), atom_map)
-        # TODO: probably vmap
-        mol_a_chiral_conflicts.append(np.array([U_a(x) for x in xs_a]))
-        mol_b_chiral_conflicts.append(np.array([U_b(x) for x in xs_b]))
+    vacuum_results = run_vacuum(
+        atom_map.mol_a,
+        atom_map.mol_b,
+        atom_map.core,
+        ff,
+        None,
+        DEFAULT_HREX_PARAMS,
+        min_overlap=0.667,  # No need to have a higher overlap then 0.667
```

Comment:
Note this significantly speeds up this test. Previously every vacuum simulation took up 48 windows

Reply:
Saves about 40 minutes of the nightly tests